### PR TITLE
Update to latest docker-java 2.1.1

### PR DESCRIPTION
**Pull request by llamahunter**
Addressing issue https://github.com/docker-java/docker-java/issues/322

### CODE REVIEW

**Comment by bmuschko** [comment]
Please also update the documentation. Any chance you can look into why the build is failing?

**Comment by llamahunter** [comment]
It's possible that the new 2.1.1 version of docker-java isn't yet in maven central.  I submitted the pull request only about 30 minutes after the docker-java release.  Hmm.. appears that 2.1.1 *still* isn't in maven central. Lemme ask the docker-java people what's up.  How do I get Snap CI to rebuild?

Which documentation files?  RELEASE_NOTES.md?

**Comment by christianhuening** [comment]
I am experiencing problems with pulling images via the DockerPullImage Task, since apparently there is no support for the Docker v1.20 API in the currently used docker-java version. (In fact the resulting REST call string is missing the "v1.20" part when looking at my docker.log).

Is this possibly going to be fixed by updating to docker-java 2.1.1 ? Or should I look further into that issue?

**Comment by bmuschko** [comment]
> It's possible that the new 2.1.1 version of docker-java isn't yet in maven central.

Ah, the joys of Maven Central. :-)

> Which documentation files? RELEASE_NOTES.md?

I mean `README.md`.

**Comment by llamahunter** [comment]
Updated Readme.asciidoc.  Hope that's what you meant.  Still not in maven central, so build still failing. :-/

File: README.asciidoc [hunk 1]
```diff
@@ -23,7 +23,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.bmuschko:gradle-docker-plugin:2.6'
+        classpath 'com.bmuschko:gradle-docker-plugin:2.6.1'
```
Comment (bmuschko) [inline README.asciidoc]: Can you remove the version change? I can do that on my end.

File: build.gradle [hunk 2]
```diff
@@ -1,5 +1,5 @@
 group = 'com.bmuschko'
-version = '2.6'
+version = '2.6.1'
```
Comment (bmuschko) [inline build.gradle]: Can you remove the version change? I can do that on my end.

File: README.asciidoc [hunk 3]
```diff
@@ -6,7 +6,7 @@ image:https://d3oypxn00j2a10.cloudfront.net/0.12.6/img/nav/docker-logo-loggedout
 Gradle plugin for managing link:https://www.docker.io/[Docker] images and containers using via its
 link:http://docs.docker.io/reference/api/docker_remote_api/[remote API]. The heavy lifting of communicating with the
 Docker remote API is handled by the link:https://github.com/docker-java/docker-java[Docker Java library]. Currently,
-version 2.1.0. Please refer to the library's documentation for more information on the supported Docker's client API
+version 2.1.1. Please refer to the library's documentation for more information on the supported Docker's client API
```
Comment (bmuschko) [inline README.asciidoc]: Thanks!

**Comment by llamahunter** [comment]
Argh... removed the wrong version bump.

**Comment by llamahunter** [comment]
Ok... now I think it's as you want it.  Except, of course, that the build fails because docker-java is having issues getting their latest release from sonatype to maven central.  They've opened a ticket about that.

**Comment by llamahunter** [comment]
docker-java 2.1.1 finally made it to maven central.  builds are now working.  when will you pull and release 2.6.1? :-)

**Comment by llamahunter** [comment]
Is there a planned 2.6.1 release date?  I would like to stop using my local SNAPSHOT build.

**Comment by bmuschko** [comment]
Probably by the end of the week. It's just a matter of time. BTW: I am looking for new maintainers for the plugin. Interested? :-)

**Comment by bmuschko** [comment]
Thanks for the pull request. It has been merged. BTW: If you need a custom version of Docker Java you can also override the default version from your build script.

    dependencies {
        docker ...
    }

**Comment by llamahunter** [comment]
I can override the dependency that's baked in to gradle-docker-plugin within my build-gradle file? Could you spell that out in a bit more detail?

**Comment by llamahunter** [comment]
In terms of maintaining the plugin... ah, not sure I have the bandwidth. But I'm sure that's a common refrain.  I'm certainly willing to contribute fixes/enhancements, tho.

**Comment by bmuschko** [comment]
See example above in my previous comment.

**Comment by llamahunter** [comment]
thx!

**Pull request merged**